lib/LedgerBlockStorage.js

```diff
@@ -185,104 +174,75 @@ class LedgerBlockStorage {
-        });
+  async getSummary({blockId, consensus = true, eventHash}) {
+    const query = {
+      id: database.hash(blockId),
+      'meta.deleted': {
+        $exists: false
       },
-      eventHashes: ['find', (results, callback) => {
-        if(!eventHash) {
-          return callback();
-        }
-        // TODO: make code DRY
-        // FIXME: this might be accomplished with aggregate query
-        // get event hashes
-        const block = results.find.block;
-        const query = {'meta.blockHeight': block.blockHeight};
-        const projection = {_id: 0, 'meta.eventHash': 1};
-        this.eventCollection.find(query, projection)
-          .sort({'meta.blockOrder': 1})
-          .toArray((err, eventHashes) => {
-            if(err) {
-              return callback(err);
-            }
-            block.eventHash = eventHashes.map(r => r.meta.eventHash);
-            callback();
-          });
-      }]
-    }, (err, results) => {
-      if(err) {
-        return callback(err);
+      'meta.consensus': {
+        $exists: true
       }
-      callback(
-        null, {block: results.find.block, meta: results.find.meta});
-    });
+    };
+    if(consensus === false) {
+      query['meta.consensus'].$exists = false;
+    }
+    const projection = {};
+    const record = await this.collection.findOne(query, projection);
+    if(!record) {
+      throw new BedrockError(
+        'A block with the given ID does not exist.',
+        'NotFoundError', {blockId});
+    }
+
+    if(eventHash) {
+      // TODO: make code DRY
+      // FIXME: this might be accomplished with aggregate query
+      // get event hashes
+      const {block} = record;
+      const query = {'meta.blockHeight': block.blockHeight};
+      const projection = {_id: 0, 'meta.eventHash': 1};
+      const records = await this.eventCollection.find(query, projection)
+        .sort({'meta.blockOrder': 1})
+        .toArray();
+      block.eventHash = records.map(r => r.meta.eventHash);
+    }
+
+    return {block: record.block, meta: record.meta};
   }
 
   /**
    * Gets a block that has consensus given a blockHeight.
    *
    * @param blockHeight - the height of the block that has consensus.
-   * @param callback(err, block) - the callback to call when finished.
-   *   err - An Error if an error occurred, null otherwise.
-   *   block - the block with the given block height that has consensus.
+   *
+   * @return a Promise that resolves to the block with the given block height
+   *   that has consensus.
    */
-  getByHeight(blockHeight, callback) {
-    async.auto({
-      find: callback => {
-        // find an existing block with consensus
-        const query = {
-          'block.blockHeight': blockHeight,
-          'meta.deleted': {
-            $exists: false
-          },
-          'meta.consensus': {
-            $exists: true
-          }
-        };
-        this.collection.findOne(query, callback);
+  async getByHeight(blockHeight) {
```

Comment:
I found it curious that the error below was not detected and it appears to me that this API is not used anywhere.  Does it seem useful?  Should it be dropped?

Reply:
It should either be tested or dropped IMO ... but not in this PR.